[Windows 10] Build error: conflicting types for SDL_main

As the title says, trying to do `stack install sdl2-image`.

```c
λ stack build
sdl2-image-2.0.0: configure
sdl2-image-2.0.0: build
Progress 1/18

--  While building custom Setup.hs for package sdl2-image-2.0.0 using:
      C:\sr\setup-exe-cache\x86_64-windows\Cabal-simple_Z6RU0evB_2.2.0.1_ghc-8.4.3.exe --builddir=.stack-work\dist\7d103d30 build --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
    Logs have been written to: C:\Users\Ashley\Desktop\haskell-sdl2-examples-master\.stack-work\logs\sdl2-image-2.0.0.log

    Configuring sdl2-image-2.0.0...
    Preprocessing library for sdl2-image-2.0.0..
    In file included from C:/Users/Ashley/AppData/Local/Programs/stack/x86_64-windows/msys2-20180531/mingw64/include/SDL2/SDL.h:32:0,
                     from C:/Users/Ashley/AppData/Local/Programs/stack/x86_64-windows/msys2-20180531/mingw64/include/SDL2/SDL_image.h:27,
                     from Image.hsc:71:
    C:/Users/Ashley/AppData/Local/Programs/stack/x86_64-windows/msys2-20180531/mingw64/include/SDL2/SDL_main.h:111:17: error: conflicting types for 'SDL_main'
     #define main    SDL_main
                     ^
    C:\Users\Ashley\AppData\Local\Temp\stack11840\sdl2-image-2.0.0\Image.hsc:73:5: note: in expansion of macro 'main'
    C:/Users/Ashley/AppData/Local/Programs/stack/x86_64-windows/msys2-20180531/mingw64/include/SDL2/SDL_main.h:117:39: note: previous declaration of 'SDL_main' was here
     extern C_LINKAGE SDLMAIN_DECLSPEC int SDL_main(int argc, char *argv[]);
                                           ^~~~~~~~
    compiling .stack-work\dist\7d103d30\build\SDL\Raw\Image_hsc_make.c failed (exit code 1)
    command was: C:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3\lib/../mingw/bin\gcc.exe -c .stack-work\dist\7d103d30\build\SDL\Raw\Image_hsc_make.c -o .stack-work\dist\7d103d30\build\SDL\Raw\Image_hsc_make.o -fno-stack-protector -D__GLASGOW_HASKELL__=804 -Dmingw32_BUILD_OS=1 -Dx86_64_BUILD_ARCH=1 -Dmingw32_HOST_OS=1 -Dx86_64_HOST_ARCH=1 -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -IC:/Users/Ashley/AppData/Local/Programs/stack/x86_64-windows/msys2-20180531/mingw64/include/SDL2 -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -IC:/Users/Ashley/AppData/Local/Programs/stack/x86_64-windows/msys2-20180531/mingw64/include/SDL2 -Dmain=SDL_main -I.stack-work\dist\7d103d30\build\autogen -I.stack-work\dist\7d103d30\build\global-autogen -include .stack-work\dist\7d103d30\build\autogen\cabal_macros.h -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:/Users/Ashley/AppData/Local/Programs/stack/x86_64-windows/msys2-20150512/mingw64/include/SDL2 -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\sr\snapshots\cf99d8ac\lib\x86_64-windows-ghc-8.4.3\vector-0.12.0.1-1fezwShzdEfG3Hp1Pmvjkq\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\sr\snapshots\cf99d8ac\lib\x86_64-windows-ghc-8.4.3\primitive-0.6.4.0-39Pwmm1zkQX6bM7xFUT3Jc\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3\lib\time-1.8.0.2\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3\lib\Win32-2.6.1.0\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3\lib\bytestring-0.10.8.2\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\msys2-20150512\mingw64\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3\lib\base-4.11.1.0\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3\lib\integer-gmp-1.0.2.0\include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3\lib/include -IC:\Users\Ashley\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.3/lib/include/
```

I uninstalled Stack from my Windows machine to try and build this using [this guide](https://www.reddit.com/r/haskellgamedev/comments/4jpthu/windows_sdl2_is_now_almost_painless_via_stack/). I also installed sdl2-image with:
```sh
λ stack exec -- pacman --noconfirm -S mingw-w64-x86_64-SDL2_gfx mingw-w64-x86_64-SDL2_image mingw-w64-x86_64-SDL2_mixer mingw-w64-x86_64-SDL2_ttf
```

I don't know how often people are using SDL2-image, or if they're just using it on something other than Windows, but without smooth compilations on Windows people aren't going to take gamedev with Haskell seriously, so let me know if there's any data you need me to gather as I've read you don't have access to a Windows machine yourself. I really want this to work as I can't really work on my Haskell game too much until I know there's a way of building it on Windows.

Thanks!

--- a/sdl2-image.cabal
+++ b/sdl2-image.cabal
@@ -43,7 +43,7 @@ library
     Haskell2010
 
   if os(windows)
-    cpp-options: -D_SDL_main_h
+    cpp-options: -D_SDL_main_h -DSDL_main_h_
 
 flag example
   description: Build the example executable

